projects/module-2/fitlit-part-one-agile.md

```diff
@@ -101,185 +105,244 @@ You should not be hardcoding the date.  You should be using code to find the las
-* 1: Application crashes during normal usage.
-
-### JavaScript / SRP / Fetch
-* 4: Application demonstrates excellent knowledge of JavaScript syntax, style, and refactoring. Application is expertly divided into logical components each with a clean, single responsibility.
-* 3: Class methods use array and object prototypes - `for` loops are not used in the application. Application shows strong effort towards organization, content, and refactoring. Application is effectively broken into logical components, but violate the principle of SRP.
-* 2: Class methods use a mix of array and object prototypes and `for` loops. Application runs but the code has long methods, unnecessary or poorly named variables, and needs significant refactoring. Divisions of logic into classes are inconsistent or unclear.
-* 1: Application generates syntax error or crashes during execution. Application logic shows poor decomposition with too much logic mashed together.
-
-### Testing
-* 4: Application covers all aspects of the application including various flows and covers both happy/sad paths. Tests must be passing to be considered.
-* 3: Application is well tested but fails to cover some features and only tests for happy paths. Tests use smaller, sample data files as input rather than the large, original data files. Some use of beforeEach to DRY up tests. Tests must be passing to be considered.
-* 2: Application makes some use of tests, but the coverage is insufficient given project requirements. No use of beforeEach to DRY up tests. Tests must be passing to be considered.
-* 1: Application does not demonstrate strong use of TDD, and some tests might be failing.
+## Testing
 
-### User Interface
-* 4: The application can stand on its own to be used by an instructor without guidance from a developer on the team. The UI does not detract from the UX. Design is responsive across small, medium and large breakpoints.  
-* 3: The application has many strong displays/interactions.The UI does not detract from the UX.
-* 2: The application shows effort in the interface, but the result is not effective. The evaluator has some difficulty using the application when reviewing the features in the users' needs.
-* 1: The application is confusing or difficult to use.
+You should *NOT* use the original data files in the `data` directory for testing. These are big files to begin with, and a real-world dataset would have millions of records. That’s far too big to use every time you want to run a test.
+
+Instead, for your tests, you should create small, sample datasets that match the structure of the application data. By creating this sample dataset, you will also know if your functions are working correctly because you can do the calculations by hand with a much smaller dataset.
+
+**You are *expected* to test:**
+- All functions that do not update the DOM.  This means everything in your `scripts.js` file that should be tested.
```

Comment:
As I worked on the testing lesson I realized some challenges that students are going to encounter with testing.  If their functions are updating a global variable, without taking it in as an argument, it's going to be really hard or impossible to test those functions.  Hopefully this will guide them towards more pure solutions but I suspect they will still end up with some impure functions that are really hard or impossible to test. 

I'm wondering what the best wording is for what we expect them to test?  "All functions that do not update the DOM."  is probably not realistic (though I used the same wording in Whats Cookin Part 1)

Reply:
Hmm good point Heather, I'll definitely think about rewording that some more!  Should there be a note of "All pure functions that do not update the DOM?"  Or "Strive to test all functions that do not update the DOM?"

Reply:
After discussing with Heather more, we discussed updating this to focus more on what they "should *NOT*" test including DOM updates and network requests.